### ISSUE TITLE
Windows 11: with `Prompt local user to accept connections` all connections fail

### ISSUE THREAD

**Issue by calestyo** [TigerVNC/tigervnc#1644]
**Describe the bug**
I've experience this for quite a while on only one system, others (with the same configuration, except for Windows being 10) work fine.

TigerVNC 1.13.1 runs on Windows 11 64 bit, as a service. It listens to local connections only. It's also configured to require confirmation on the service side (by clicking the accept button). VNC auth is none, encryption is none... but I also tried all combinations with password and anonymous TLS.<br>
Cygwin's sshd is running.

From some Linux I do a port forwarding via SSH, so before any VNC client even tries to connect, the connection SSH is already open, thus it cannot be the Windows Firewall.

Now when I connect via a VNC client (I've tried remmina 1.4.31 and TigerVNC 1.12.0 - just to rule out that it's a problem in remmina), nothing happens on the computer running the server (i.e. there is no pop up that asks for acceptance or rejection) and the client times out.<br>
Trying the same again ... and it works (from then on).

**Client (please complete the following information):**
 - OS: Debian sid
 - VNC client: TigerVNC (1.12.0), remmina (1.4.13)
 - Client downloaded from: Debian sid

**Server (please complete the following information):**
 - OS: Windows 11
 - VNC server: TigerVNC
 - VNC server version: 1.13.0 (but also happened wit 1.12.x)
 - Server downloaded from: tigervnc.ogr
 - Server was started using: as the Windows service

Any ideas?

**Comment by CendioOssman** [comment]
Nothing obvious comes to mind. Could you get some debug logs from the server and client when this happens? We have instructions in the wiki for how to get such logs.

**Comment by calestyo** [comment]
Done as described [here](https://github.com/TigerVNC/tigervnc/wiki/Debug-Logs). For the server on Windows, I've modified the Windows service to include the debug options.

The server's log is:
[WinVNC4.log](https://github.com/TigerVNC/tigervnc/files/11982564/WinVNC4.log)

For the client (Linux) I've used again TigerVNC 1.12.0.<br>
stdout was empty for both cases (the first connection, that times out with no pop up on the server, and the 2nd which then works.

stderr was:
```
$ vncviewer -Log *:stderr:100 localhost:51000 > tigervnc-viewer.log

TigerVNC Viewer 64-bit v1.12.0
Built on: 2023-01-06 16:01
Copyright (C) 1999-2021 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.
XOpenIM() failed

Fri Jul  7 15:05:11 2023
 DecodeManager: Detected 16 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 TcpSocket:   Connecting to localhost [::1] port 51000
 CConn:       Connected to host localhost port 51000
 CConnection: reading protocol version
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: processing security types message
 CConnection: processing security types message
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type None(1)
 CConnection: Choosing security type VeNCrypt(19)
 CConnection: processing security message
 CConnection: processing security message
 CConnection: processing security message
 CVeNCrypt:   Server offers security type None (1)
 CVeNCrypt:   Choosing security type None (1)
 CConnection: processing security result message

Fri Jul  7 15:05:22 2023
 CConnection: processing security result message
 CConnection: auth failed
 CConnection: processing security reason message
 CConn:       Authentication failure: Connection rejected by user
```
for the first (failed) connection.<br>
And for the 2nd:
```
$ vncviewer -Log *:stderr:100 localhost:51000 > tigervnc-viewer.log2

TigerVNC Viewer 64-bit v1.12.0
Built on: 2023-01-06 16:01
Copyright (C) 1999-2021 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.
XOpenIM() failed

Fri Jul  7 15:05:36 2023
 DecodeManager: Detected 16 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 TcpSocket:   Connecting to localhost [::1] port 51000
 CConn:       Connected to host localhost port 51000
 CConnection: reading protocol version
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: processing security types message
 CConnection: processing security types message
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type None(1)
 CConnection: Choosing security type VeNCrypt(19)
 CConnection: processing security message
 CConnection: processing security message
 CConnection: processing security message
 CVeNCrypt:   Server offers security type None (1)
 CVeNCrypt:   Choosing security type None (1)
 CConnection: processing security result message

Fri Jul  7 15:05:39 2023
 CConnection: processing security result message
 CConnection: Authentication success!
 CConnection: reading server initialisation
 CConnection: reading server initialisation
 CConnection: initialisation done
 Viewport:    No key mapping for key LVL3
 Viewport:    No key mapping for key I178
 Viewport:    No key mapping for key I182
 Viewport:    No key mapping for key I183
 Viewport:    No key mapping for key LVL5
 Viewport:    No key mapping for key ALT
 Viewport:    No key mapping for key META
 Viewport:    No key mapping for key SUPR
 Viewport:    No key mapping for key HYPR
 Viewport:    No key mapping for key I248
 Viewport:    No key mapping for key I249
 Viewport:    No key mapping for key I250
 Viewport:    No key mapping for key I251
 Viewport:    No key mapping for key I252
 Viewport:    No key mapping for key I253
 Viewport:    No key mapping for key I254
 PlatformPixelBuffer: Using shared memory XImage
 DesktopWindow: Adjusting window size to avoid accidental full screen request
 CConn:       Using pixel format depth 24 (32bpp) little-endian rgb888
 CMsgHandler: Got server clipboard capabilities:
 CMsgHandler:     Plain text (only notify)
 CConnection: Enabling continuous updates
 DesktopWindow: Requesting framebuffer resize from 1920x1080 to 1920x1003
 DesktopWindow: 1 screen(s)
 DesktopWindow:              0 (0x00000000): 1920x1003+0+0 (flags 0x00000000)
 DesktopWindow: 
 CConn:       SetDesktopSize failed: 1

Fri Jul  7 15:05:53 2023
 PlatformPixelBuffer: Freeing shared memory XImage
```

Thanks :-)

**Comment by CendioOssman** [comment]
Hmm... Something didn't quite take there, as the server log is almost empty.

Still, the client message indicates that there is something wrong with the query mechanism. I'm fairly sure we had another report about that and WinVNC some time ago, but I can't find it.

We have had some fixes in the connection handling since release. Could you test a nightly build and see if it behaves the same?

**Comment by calestyo** [comment]
First, there was some mess up... apparently the server was still at 1.12.0... (I guess the `.exe` file names on the sf download confused me... would perhaps be better if the client files could be named `…-client` and the server `…-server` and not (`…winvnc…` :-D).

Anyway, repeated it now with 1.13.1 but same result.

Also, what I've noted previously but forgot to mention, but which did **not** happen when trying today, is:
At previous tries, when clicking on the TigerVNC systray icon and there "About"… nothing happened, i.e. no window popped up. But today that worked.

As for the logs, that's all that was written, and I guess that was with the maximum verbosity?

Where can one get nightly builds for Windows? SF seems to only have the stable ones?

Speaking of which,… wasn't SF at some point adding some undesired stuff? Couldn't the builds be provided via Github itself?

**Comment by CendioOssman** [comment]
> As for the logs, that's all that was written, and I guess that was with the maximum verbosity?

Something is amiss because we've had other WinVNC reports with way more details. There should be lots more for each connection.

> Where can one get nightly builds for Windows? SF seems to only have the stable ones?

It's on the home page, hidden in the Downloads section. :)

> Speaking of which,… wasn't SF at some point adding some undesired stuff?

Yup. We switched to GitHub as part of that debacle.

> Couldn't the builds be provided via Github itself?

Unfortunately, no. Their interface is far too primitive to host the number of files we have in a release. We used bintray for a while, but they went away. And meanwhile, SourceForge decided to clean up their act. So we're back there for now.

**Comment by calestyo** [comment]
Hey @CendioOssman.

I just tried that again (with the stable version, http://tigervnc.bphinz.com/nightly/ seems broken) in all possible combinations.

With polling/with hooks. With auth none, with password, with my SSH tunnel using on the TigerVNC side `::1` or `127.0.0.1` etc. pp. - nothing really changes the problem.

The first connection **does** work, when the prompt for connection confirmation is disabled.

If not, then the first connection after the service is started (or re-started) times out at the client side, as no connection prompt shows up on the server side.<br>
But what I do see even on first connections is, that the background window disappears, so TigerVNC does get the connection it just doesn't show the confirmation popup.

Also it's really enough to re-trigger the behaviour by stopping/starting the service again. Even when the same SSH port forwarding tunnel is still open.

Might be just a GUI issue.

Cheers,
Chris.

**Comment by bphinz** [comment]
> http://tigervnc.bphinz.com/nightly/ seems broken

Thanks for noticing that.  It looks like one of the raspberry pi build
hosts has been down, and in the meantime the S2 build artifacts expired.
I'll get that back online today

**Comment by calestyo** [comment]
Some update on this.

First, already with 1.13.1, the first connection fails only if `Prompt local user to accept connections` is enabled (regardless of whether `Only prompt when there is a user logged on` is enabled or not.<br>
As mentioned previously, there is some form of connection (i.e. on the server, the Windows background gets disabled) but there is simply no pop up asking for accepting/rejecting the connection.
<hr>
I now tried again the fresh version 1.14.0 (i.e. the server), on both Windows 10 and Windows 11.<br>
Problem got worse there.

Now, not only the first connection fails, but also any subsequent ones. Again only when `Prompt local user to accept connections` is enabled. Again, with remmina as well as with TigerVNC viewer. (these both from Debian unstable)

The error message have changed, though.<br>
With 1.13.1, the error message was something like `Connection rejected by user` (for both remmina and TigerVNC viewer).<br>
With 1.14.0, the error message depends on whether session encryption is `None` or `Anonymous TLS`:

With `Anonymous TLS` and `Prompt local user to accept connections` disabled (connection works):
```
$ xtigervncviewer localhost::51000 -Log *:stderr:100

TigerVNC Viewer v1.13.1
Built on: 2024-07-29 12:35
Copyright (C) 1999-2022 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.
XOpenIM() failed

Wed Jul 31 18:42:00 2024
 DecodeManager: Detected 16 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 TcpSocket:   Connecting to localhost [::1] port 51000
 CConn:       Connected to host localhost port 51000
 CConnection: reading protocol version
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: processing security types message
 CConnection: processing security types message
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type None(1)
 CConnection: Choosing security type VeNCrypt(19)
 CConnection: processing security message
 CConnection: processing security message
 CConnection: processing security message
 CVeNCrypt:   Server offers security type TLSNone (257)
 CVeNCrypt:   Server offers security type None (1)
 CVeNCrypt:   Choosing security type TLSNone (257)
 CConnection: processing security message
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 CConnection: processing security message
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 CConnection: processing security message
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)
 CConnection: processing security result message
 CConnection: Authentication success!
 CConnection: reading server initialisation
 CConnection: reading server initialisation
 CConnection: initialisation done
 Viewport:    No key mapping for key LVL3
 Viewport:    No key mapping for key I178
 Viewport:    No key mapping for key I182
 Viewport:    No key mapping for key I183
 Viewport:    No key mapping for key LVL5
 Viewport:    No key mapping for key ALT
 Viewport:    No key mapping for key META
 Viewport:    No key mapping for key SUPR
 Viewport:    No key mapping for key HYPR
 Viewport:    No key mapping for key I248
 Viewport:    No key mapping for key I249
 Viewport:    No key mapping for key I250
 Viewport:    No key mapping for key I251
 Viewport:    No key mapping for key I252
 Viewport:    No key mapping for key I253
 Viewport:    No key mapping for key I254
 PlatformPixelBuffer: Using shared memory XImage
 DesktopWindow: Reducing window size to fit on current monitor
 CConn:       Using pixel format depth 24 (32bpp) little-endian rgb888
 CMsgHandler: Got server clipboard capabilities:
 CMsgHandler:     Plain text (only notify)
 Viewport:    Got server LED state: 0x00000002
 CConnection: Enabling continuous updates
 DesktopWindow: Requesting framebuffer resize from 1920x1080 to 1920x1003
 DesktopWindow: 1 screen(s)
 DesktopWindow:              0 (0x00000000): 1920x1003+0+0 (flags 0x00000000)
 DesktopWindow: 
 Viewport:    Inserting fake NumLock to get in sync with server
 Viewport:    Key pressed: 0x0045 => XK_Num_Lock (0xff7f)
 Viewport:    Key released: 0x0045 => XK_Num_Lock (0xff7f)
 CConn:       SetDesktopSize failed: 1

Wed Jul 31 18:42:10 2024
 PlatformPixelBuffer: Freeing shared memory XImage
 DecodeManager:     raw: 2 rects, 228 pixels
 DecodeManager:          936 B (1:1 ratio)
 DecodeManager:     Tight: 219 rects, 2,51843 Mpixels
 DecodeManager:            374,198 KiB (1:26,2967 ratio)
 DecodeManager:   Total: 221 rects, 2,51866 Mpixels
 DecodeManager:          375,112 KiB (1:26,2351 ratio)
```

With `Anonymous TLS` and `Prompt local user to accept connections` enabled (connection does **not** work):
```
$ xtigervncviewer localhost::51000 -Log *:stderr:100

TigerVNC Viewer v1.13.1
Built on: 2024-07-29 12:35
Copyright (C) 1999-2022 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.
XOpenIM() failed

Wed Jul 31 18:40:16 2024
 DecodeManager: Detected 16 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 TcpSocket:   Connecting to localhost [::1] port 51000
 CConn:       Connected to host localhost port 51000
 CConnection: reading protocol version
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: processing security types message
 CConnection: processing security types message
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type None(1)
 CConnection: Choosing security type VeNCrypt(19)
 CConnection: processing security message
 CConnection: processing security message
 CConnection: processing security message
 CVeNCrypt:   Server offers security type TLSNone (257)
 CVeNCrypt:   Server offers security type None (1)
 CVeNCrypt:   Choosing security type TLSNone (257)
 CConnection: processing security message
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 CConnection: processing security message
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.

Wed Jul 31 18:40:19 2024
 CConnection: processing security message
 TLS:         TLS Handshake failed: The TLS connection was non-properly
              terminated.
 TLS:         
 CConn:       Authentication failure: TLS Handshake failed
 DecodeManager:   Total: 0 rects, 0 pixels
 DecodeManager:          0 B (1:-nan ratio)
```
So the error is now some "TLS Handshake failed: The TLS connection was non-properly terminated."

With `None` and `Prompt local user to accept connections` disabled (connection works):
```
$ xtigervncviewer localhost::51000 -Log *:stderr:100 -SecurityTypes None

TigerVNC Viewer v1.13.1
Built on: 2024-07-29 12:35
Copyright (C) 1999-2022 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.

Wed Jul 31 18:43:22 2024
 Config:      set SecurityTypes(String) to None
XOpenIM() failed
 DecodeManager: Detected 16 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 TcpSocket:   Connecting to localhost [::1] port 51000
 CConn:       Connected to host localhost port 51000
 CConnection: reading protocol version
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: processing security types message
 CConnection: processing security types message
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type None(1)
 CConnection: Choosing security type None(1)
 CConnection: processing security message
 CConnection: processing security result message
 CConnection: processing security result message
 CConnection: Authentication success!
 CConnection: reading server initialisation
 CConnection: reading server initialisation
 CConnection: initialisation done
 Viewport:    No key mapping for key LVL3
 Viewport:    No key mapping for key I178
 Viewport:    No key mapping for key I182
 Viewport:    No key mapping for key I183
 Viewport:    No key mapping for key LVL5
 Viewport:    No key mapping for key ALT
 Viewport:    No key mapping for key META
 Viewport:    No key mapping for key SUPR
 Viewport:    No key mapping for key HYPR
 Viewport:    No key mapping for key I248
 Viewport:    No key mapping for key I249
 Viewport:    No key mapping for key I250
 Viewport:    No key mapping for key I251
 Viewport:    No key mapping for key I252
 Viewport:    No key mapping for key I253
 Viewport:    No key mapping for key I254
 PlatformPixelBuffer: Using shared memory XImage

Wed Jul 31 18:43:23 2024
 DesktopWindow: Reducing window size to fit on current monitor
 CConn:       Using pixel format depth 24 (32bpp) little-endian rgb888
 CMsgHandler: Got server clipboard capabilities:
 CMsgHandler:     Plain text (only notify)
 Viewport:    Got server LED state: 0x00000002
 Viewport:    Inserting fake NumLock to get in sync with server
 Viewport:    Key pressed: 0x0045 => XK_Num_Lock (0xff7f)
 Viewport:    Key released: 0x0045 => XK_Num_Lock (0xff7f)
 CConnection: Enabling continuous updates
 DesktopWindow: Requesting framebuffer resize from 1920x1080 to 1920x1003
 DesktopWindow: 1 screen(s)
 DesktopWindow:              0 (0x00000000): 1920x1003+0+0 (flags 0x00000000)
 DesktopWindow: 
 CConn:       SetDesktopSize failed: 1

Wed Jul 31 18:43:25 2024
 PlatformPixelBuffer: Freeing shared memory XImage
 DecodeManager:     raw: 2 rects, 228 pixels
 DecodeManager:          936 B (1:1 ratio)
 DecodeManager:     Tight: 93 rects, 2,20896 Mpixels
 DecodeManager:            347,394 KiB (1:24,8417 ratio)
 DecodeManager:   Total: 95 rects, 2,20919 Mpixels
 DecodeManager:          348,308 KiB (1:24,7791 ratio)
```

With `None` and `Prompt local user to accept connections` enabled (connection does **not** work):
```
$ xtigervncviewer localhost::51000 -Log *:stderr:100 -SecurityTypes None

TigerVNC Viewer v1.13.1
Built on: 2024-07-29 12:35
Copyright (C) 1999-2022 TigerVNC Team and many others (see README.rst)
See https://www.tigervnc.org for information on TigerVNC.

Wed Jul 31 18:44:00 2024
 Config:      set SecurityTypes(String) to None
XOpenIM() failed
 DecodeManager: Detected 16 CPU core(s)
 DecodeManager: Creating 4 decoder thread(s)
 TcpSocket:   Connecting to localhost [::1] port 51000
 CConn:       Connected to host localhost port 51000
 CConnection: reading protocol version
 CConnection: Server supports RFB protocol version 3.8
 CConnection: Using RFB protocol version 3.8
 CConnection: processing security types message
 CConnection: processing security types message
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type VeNCrypt(19)
 CConnection: Server offers security type None(1)
 CConnection: Choosing security type None(1)
 CConnection: processing security message
 CConnection: processing security result message

Wed Jul 31 18:44:02 2024
 CConnection: processing security result message
 CConn:       End of stream
 CConn:       The connection was dropped by the server before the session could
              be established.
 DecodeManager:   Total: 0 rects, 0 pixels
 DecodeManager:          0 B (1:-nan ratio)
```
so no longer a user-rejected error as in 1.13.1, but "The connection was dropped by the server before the session could be established."


Cheers,
Chris.

**Comment by calestyo** [comment]
Tried again with 1.15.0, regardless of authentication methods and session encryption,... the connection is immediately closed after the passphrase has been entered.

On the windows side one sees a small flickering of the screen (which is - I guess - because the VNC server shortly disables the wallpaper, but immediately brings it back).<br>
The pop-up where the user on the VNC server side would be asked whether the connection is accepted never shows up.
Happens on every login try, i.e. no longer (as in 1.13.1) that it doesn't only work on the first connection, but does work on subsequent ones.

Happens with remmina an tigervnc-viewer. Also tried both IPv4 and v6 (though it was extremely unlikely to be the cause).

**Comment by CendioOssman** [comment]
Have you made a new attempt at getting a server log? Everything points to a server issue, so a complete debug log there is crucial to debug this.

**Comment by calestyo** [comment]
Here's a debug log with 1.15.0:

Starting the server (session encryption is None and Anon TLS:
```
$ tail -f WinVNC4.log 
 RegConfig:   Unable to process UseHooks
 Config:      Set Protocol3.3(Bool) to off
 Config:      Set Password(Binary)
 ManagedListener: Set server to 0000007b46fff580
 ManagedListener: Set port to 5900
 ManagedListener: Set filter to +,
 VNCServerWin32: Refreshing tray icon
 VNCServerWin32: Entering message loop
 MsgWindow:   Null _this in 0000000000090710, message 83
 MsgWindow:   Created window "VNCTray" (0000000000090710)
```

Now I change authn to Standard VNC only:
```
Wed Jun 04 18:03:40 2025
 RegConfig:   Registry changed
 Config:      Set SecurityTypes(String) to VeNCrypt,TLSVnc,VncAuth
 Config:      Set QueryConnect(Bool) to on
 Config:      Set QueryOnlyIfLoggedOn(Bool) to off
 Config:      Set PortNumber(Int) to 5900
 Config:      Set IdleTimeout(Int) to 0
 Config:      Set LocalHost(Bool) to on
 Config:      Set Hosts(String) to +,
 Config:      Set AcceptKeyEvents(Bool) to on
 Config:      Set RawKeyboard(Bool) to off
 Config:      Set AcceptPointerEvents(Bool) to on
 Config:      Set AcceptCutText(Bool) to off
 Config:      Set SendCutText(Bool) to off
 Config:      Set DisableLocalInputs(Bool) to off
 Config:      Set DisconnectClients(Bool) to on
 Config:      Set AlwaysShared(Bool) to off
 Config:      Set NeverShared(Bool) to off
 Config:      Set DisconnectAction(String) to None
 Config:      Set RemoveWallpaper(Bool) to on
 Config:      Set DisableEffects(Bool) to on
 Config:      Set UpdateMethod(Int) to 0
 Config:      Set PollConsoleWindows(Bool) to on
 Config:      Set UseCaptureBlt(Bool) to on
 RegConfig:   Unable to process UseHooks
 Config:      Set Protocol3.3(Bool) to off
 Config:      Set Password(Binary)
 ManagedListener: Set filter to +,
 VNCServerWin32: Refreshing tray icon
```
and connect with remmina, which gives me the passphrase prompt and after entering that, the Windows screen "flickrs" shortly, but gives no prompt whether the client shall be accepted:
```
Wed Jun 04 18:03:51 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::55637
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSVnc (258)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db74e5d20)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db74e5d20)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)

Wed Jun 04 18:03:57 2025
 SConnection: Processing security message
 VNCServerST: Starting desktop
 SDisplay:    Starting
 SDisplay:    Attaching to virtual desktop
 SDisplay:    Deleting old pixel buffer & device
 SDisplay:    Creating pixel buffer
 DeviceContext: 32-bit BitFields: (ff0000, ff00, ff)
 DIBSectionBuffer: recreateBuffer()

Wed Jun 04 18:03:58 2025
 MsgWindow:   Creating window "rfb::win32::SDisplayCorePolling"
 MsgWindow:   Null _this in 0000000000160658, message 24
 MsgWindow:   Null _this in 0000000000160658, message 81
 MsgWindow:   Null _this in 0000000000160658, message 83
 MsgWindow:   Created window "rfb::win32::SDisplayCorePolling"
              (0000000000160658)
 SDisplayCorePolling: setScreenRect
 SDisplay:    Started Polling
 MsgWindow:   Creating window "WMMonitor"
 MsgWindow:   Null _this in 00000000000c0720, message 24
 MsgWindow:   Null _this in 00000000000c0720, message 81
 MsgWindow:   Null _this in 00000000000c0720, message 83
 MsgWindow:   Created window "WMMonitor" (00000000000c0720)
 MsgWindow:   Creating window "Clipboard"
 MsgWindow:   Null _this in 00000000000c072c, message 24
 MsgWindow:   Null _this in 00000000000c072c, message 81
 MsgWindow:   Null _this in 00000000000c072c, message 83
 MsgWindow:   Created window "Clipboard" (00000000000c072c)
 Clipboard:   Local clipboard changed by 0000000000000000
 Clipboard:   No clipboard notifier registered
 Clipboard:   Registered clipboard handler
 SInput:      Found dead key 0x60 '`'
 SInput:      Found dead key 0xb4 '�'
 SInput:      Found dead key 0x5e '^'
 CleanDesktop: Disable desktop wallpaper/Active Desktop
 CleanDesktop: ActiveDesktop::enable
 CleanDesktop: Failed to get/set Active Desktop options: -2147467263
 CleanDesktop: Disable desktop effects
 SDisplay:    Started
 VNCSConnST:  Closing [::1]::55637: basic_string::_M_construct null not valid
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 Connections: Closed: [::1]::55637
 VNCServerST: Stopping desktop
 SDisplay:    Stopping
 ComparingUpdateTracker: 0 pixels in / 0 pixels out
 ComparingUpdateTracker: (1:nan ratio)
 SDisplay:    Stopping Polling
 MsgWindow:   Null _this in 0000000000160658, message 2
 MsgWindow:   Null _this in 0000000000160658, message 82
 MsgWindow:   Destroyed window "rfb::win32::SDisplayCorePolling"
              (0000000000160658)
 MsgWindow:   Null _this in 00000000000c0720, message 2
 MsgWindow:   Null _this in 00000000000c0720, message 82
 MsgWindow:   Destroyed window "WMMonitor" (00000000000c0720)
 Clipboard:   Removing 00000000000c072c from chain (next is 0000000000000000)
 MsgWindow:   Null _this in 00000000000c072c, message 2
 MsgWindow:   Null _this in 00000000000c072c, message 82
 MsgWindow:   Destroyed window "Clipboard" (00000000000c072c)
 CleanDesktop: Restore desktop effects
 Registry:    Duplicated ffffffff80000001 to ffffffff80000001
 Registry:    openKey(ffffffff80000001,Control Panel\Desktop,rw) =
              0000000000000344
 Registry:    RegCloseKey(ffffffff80000001)
 Registry:    RegCloseKey(0000000000000344)
 CleanDesktop: Restore desktop wallpaper
 SDisplay:    Stopped

Wed Jun 04 18:04:00 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::55639
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSVnc (258)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db74d7370)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db74d7370)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)

Wed Jun 04 18:04:21 2025
 SConnection: Processing security message
 VNCSConnST:  Closing [::1]::55639: readTLS: The TLS connection was
              non-properly terminated. (-110)
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 Connections: Closed: [::1]::55639
```

Changed the authn config to disable Standard VNC and enable None:
```
Wed Jun 04 18:04:32 2025
 RegConfig:   Registry changed
 Config:      Set SecurityTypes(String) to VeNCrypt,TLSNone,None
 Config:      Set QueryConnect(Bool) to on
 Config:      Set QueryOnlyIfLoggedOn(Bool) to off
 Config:      Set PortNumber(Int) to 5900
 Config:      Set IdleTimeout(Int) to 0
 Config:      Set LocalHost(Bool) to on
 Config:      Set Hosts(String) to +,
 Config:      Set AcceptKeyEvents(Bool) to on
 Config:      Set RawKeyboard(Bool) to off
 Config:      Set AcceptPointerEvents(Bool) to on
 Config:      Set AcceptCutText(Bool) to off
 Config:      Set SendCutText(Bool) to off
 Config:      Set DisableLocalInputs(Bool) to off
 Config:      Set DisconnectClients(Bool) to on
 Config:      Set AlwaysShared(Bool) to off
 Config:      Set NeverShared(Bool) to off
 Config:      Set DisconnectAction(String) to None
 Config:      Set RemoveWallpaper(Bool) to on
 Config:      Set DisableEffects(Bool) to on
 Config:      Set UpdateMethod(Int) to 0
 Config:      Set PollConsoleWindows(Bool) to on
 Config:      Set UseCaptureBlt(Bool) to on
 RegConfig:   Unable to process UseHooks
 Config:      Set Protocol3.3(Bool) to off
 Config:      Set Password(Binary)
 ManagedListener: Set filter to +,
 VNCServerWin32: Refreshing tray icon
 RegConfig:   Registry changed
 Config:      Set SecurityTypes(String) to VeNCrypt,TLSNone,None
 Config:      Set QueryConnect(Bool) to on
 Config:      Set QueryOnlyIfLoggedOn(Bool) to off
 Config:      Set PortNumber(Int) to 5900
 Config:      Set IdleTimeout(Int) to 0
 Config:      Set LocalHost(Bool) to on
 Config:      Set Hosts(String) to +,
 Config:      Set AcceptKeyEvents(Bool) to on
 Config:      Set RawKeyboard(Bool) to off
 Config:      Set AcceptPointerEvents(Bool) to on
 Config:      Set AcceptCutText(Bool) to off
 Config:      Set SendCutText(Bool) to off
 Config:      Set DisableLocalInputs(Bool) to off
 Config:      Set DisconnectClients(Bool) to on
 Config:      Set AlwaysShared(Bool) to off
 Config:      Set NeverShared(Bool) to off
 Config:      Set DisconnectAction(String) to None
 Config:      Set RemoveWallpaper(Bool) to on
 Config:      Set DisableEffects(Bool) to on
 Config:      Set UpdateMethod(Int) to 0
 Config:      Set PollConsoleWindows(Bool) to on
 Config:      Set UseCaptureBlt(Bool) to on
 RegConfig:   Unable to process UseHooks
 Config:      Set Protocol3.3(Bool) to off
 Config:      Set Password(Binary)
 ManagedListener: Set filter to +,
 VNCServerWin32: Refreshing tray icon
```
and another try with Remmina:
```
Wed Jun 04 18:04:43 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::55642
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSNone (257)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.

Wed Jun 04 18:04:44 2025
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db74c5c40)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db74c5c40)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)
 VNCServerST: Starting desktop
 SDisplay:    Starting
 SDisplay:    Attaching to virtual desktop
 SDisplay:    Deleting old pixel buffer & device
 SDisplay:    Creating pixel buffer
 DeviceContext: 32-bit BitFields: (ff0000, ff00, ff)
 DIBSectionBuffer: recreateBuffer()
 MsgWindow:   Creating window "rfb::win32::SDisplayCorePolling"
 MsgWindow:   Null _this in 00000000001605f4, message 24
 MsgWindow:   Null _this in 00000000001605f4, message 81
 MsgWindow:   Null _this in 00000000001605f4, message 83
 MsgWindow:   Created window "rfb::win32::SDisplayCorePolling"
              (00000000001605f4)
 SDisplayCorePolling: setScreenRect
 SDisplay:    Started Polling
 MsgWindow:   Creating window "WMMonitor"
 MsgWindow:   Null _this in 00000000000b071e, message 24
 MsgWindow:   Null _this in 00000000000b071e, message 81
 MsgWindow:   Null _this in 00000000000b071e, message 83
 MsgWindow:   Created window "WMMonitor" (00000000000b071e)
 MsgWindow:   Creating window "Clipboard"
 MsgWindow:   Null _this in 0000000000160726, message 24
 MsgWindow:   Null _this in 0000000000160726, message 81
 MsgWindow:   Null _this in 0000000000160726, message 83
 MsgWindow:   Created window "Clipboard" (0000000000160726)
 Clipboard:   Local clipboard changed by 0000000000000000
 Clipboard:   No clipboard notifier registered
 Clipboard:   Registered clipboard handler
 SInput:      Found dead key 0x60 '`'
 SInput:      Found dead key 0xb4 '�'
 SInput:      Found dead key 0x5e '^'
 CleanDesktop: Disable desktop wallpaper/Active Desktop
 CleanDesktop: ActiveDesktop::enable
 CleanDesktop: Failed to get/set Active Desktop options: -2147467263
 CleanDesktop: Disable desktop effects
 SDisplay:    Started
 VNCSConnST:  Closing [::1]::55642: basic_string::_M_construct null not valid
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 Connections: Closed: [::1]::55642
 VNCServerST: Stopping desktop
 SDisplay:    Stopping
 ComparingUpdateTracker: 0 pixels in / 0 pixels out
 ComparingUpdateTracker: (1:nan ratio)
 SDisplay:    Stopping Polling
 MsgWindow:   Null _this in 00000000001605f4, message 2
 MsgWindow:   Null _this in 00000000001605f4, message 82
 MsgWindow:   Destroyed window "rfb::win32::SDisplayCorePolling"
              (00000000001605f4)
 MsgWindow:   Null _this in 00000000000b071e, message 2
 MsgWindow:   Null _this in 00000000000b071e, message 82
 MsgWindow:   Destroyed window "WMMonitor" (00000000000b071e)
 Clipboard:   Removing 0000000000160726 from chain (next is 0000000000000000)
 MsgWindow:   Null _this in 0000000000160726, message 2
 MsgWindow:   Null _this in 0000000000160726, message 82
 MsgWindow:   Destroyed window "Clipboard" (0000000000160726)
 CleanDesktop: Restore desktop effects
 Registry:    Duplicated ffffffff80000001 to ffffffff80000001
 Registry:    openKey(ffffffff80000001,Control Panel\Desktop,rw) =
              00000000000003a8
 Registry:    RegCloseKey(ffffffff80000001)
 Registry:    RegCloseKey(00000000000003a8)
 CleanDesktop: Restore desktop wallpaper
 SDisplay:    Stopped
```
and for the fun one more (because it 1.13.1 it works at least after the first try on a freshly started server):
```
Wed Jun 04 18:04:58 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::55643
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSNone (257)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db6f70140)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000026db6f70140)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)
 VNCServerST: Starting desktop
 SDisplay:    Starting
 SDisplay:    Attaching to virtual desktop
 SDisplay:    Deleting old pixel buffer & device
 SDisplay:    Creating pixel buffer
 DeviceContext: 32-bit BitFields: (ff0000, ff00, ff)
 DIBSectionBuffer: recreateBuffer()
 MsgWindow:   Creating window "rfb::win32::SDisplayCorePolling"
 MsgWindow:   Null _this in 0000000000130714, message 24
 MsgWindow:   Null _this in 0000000000130714, message 81
 MsgWindow:   Null _this in 0000000000130714, message 83
 MsgWindow:   Created window "rfb::win32::SDisplayCorePolling"
              (0000000000130714)
 SDisplayCorePolling: setScreenRect
 SDisplay:    Started Polling
 MsgWindow:   Creating window "WMMonitor"
 MsgWindow:   Null _this in 00000000000906ae, message 24
 MsgWindow:   Null _this in 00000000000906ae, message 81
 MsgWindow:   Null _this in 00000000000906ae, message 83
 MsgWindow:   Created window "WMMonitor" (00000000000906ae)
 MsgWindow:   Creating window "Clipboard"
 MsgWindow:   Null _this in 00000000001706e0, message 24
 MsgWindow:   Null _this in 00000000001706e0, message 81
 MsgWindow:   Null _this in 00000000001706e0, message 83
 MsgWindow:   Created window "Clipboard" (00000000001706e0)
 Clipboard:   Local clipboard changed by 0000000000000000
 Clipboard:   No clipboard notifier registered
 Clipboard:   Registered clipboard handler
 SInput:      Found dead key 0x60 '`'
 SInput:      Found dead key 0xb4 '�'
 SInput:      Found dead key 0x5e '^'
 CleanDesktop: Disable desktop wallpaper/Active Desktop
 CleanDesktop: ActiveDesktop::enable
 CleanDesktop: Failed to get/set Active Desktop options: -2147467263
 CleanDesktop: Disable desktop effects
 SDisplay:    Started
 VNCSConnST:  Closing [::1]::55643: basic_string::_M_construct null not valid
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 Connections: Closed: [::1]::55643
 VNCServerST: Stopping desktop
 SDisplay:    Stopping
 ComparingUpdateTracker: 0 pixels in / 0 pixels out
 ComparingUpdateTracker: (1:nan ratio)
 SDisplay:    Stopping Polling
 MsgWindow:   Null _this in 0000000000130714, message 2
 MsgWindow:   Null _this in 0000000000130714, message 82
 MsgWindow:   Destroyed window "rfb::win32::SDisplayCorePolling"
              (0000000000130714)
 MsgWindow:   Null _this in 00000000000906ae, message 2
 MsgWindow:   Null _this in 00000000000906ae, message 82
 MsgWindow:   Destroyed window "WMMonitor" (00000000000906ae)
 Clipboard:   Removing 00000000001706e0 from chain (next is 0000000000000000)
 MsgWindow:   Null _this in 00000000001706e0, message 2
 MsgWindow:   Null _this in 00000000001706e0, message 82
 MsgWindow:   Destroyed window "Clipboard" (00000000001706e0)
 CleanDesktop: Restore desktop effects
 Registry:    Duplicated ffffffff80000001 to ffffffff80000001
 Registry:    openKey(ffffffff80000001,Control Panel\Desktop,rw) =
              00000000000003a8
 Registry:    RegCloseKey(ffffffff80000001)
 Registry:    RegCloseKey(00000000000003a8)
 CleanDesktop: Restore desktop wallpaper
 SDisplay:    Stopped
```

This was now in server and not in service mode,... but it seems both behave the same.


Do you know at which place in the code the windows server opens the acceptance prompt?


Cheers,
Chris.

**Comment by CendioOssman** [comment]
> Here's a debug log with 1.15.0:

Thanks. We now have an error that explains why the connection is dropped:

```
 VNCSConnST:  Closing [::1]::55637: basic_string::_M_construct null not valid
```

I seem to recall some fixes on this theme. Could you test a development version and see if it behaves better? And if not, does it still have this log line?

> Do you know at which place in the code the windows server opens the acceptance prompt?

It's done here:

https://github.com/TigerVNC/tigervnc/blob/5bfcb7a0c7559c3f029da68d89619e8944021fa8/win/winvnc/VNCServerWin32.cxx#L256-L269

**Comment by calestyo** [comment]
> Could you test a development version and see if it behaves better? And if not, does it still have this log line?

Sure... would it be feasible to provide it as like a alpha-test release or so? With the usual digital signature for Windows executables? That way it would at lest not be less trustworthy than the regular releases ;-)

**Comment by bphinz** [comment]
The nightly builds are created and signed in exactly the same way as the
release builds.

**Comment by calestyo** [comment]
With `tigervnc64-winvnc-1.15.80.exe` (again server mode)..

Which btw. seems to have another regression... I cannot change the session encryption (which always reverts to None and anon TLS) nor the authn method (which always reverts to password only) anymore - both, server and service mode.

```
Sun Jun 08 02:39:20 2025
 VNCServerST: Creating single-threaded server FERMI
 Registry:    Duplicated ffffffff80000001 to ffffffff80000001
 Registry:    createKey(ffffffff80000001,Software\TigerVNC\WinVNC4) =
              0000000000000254
 Registry:    RegCloseKey(ffffffff80000001)
 MsgWindow:   Creating window "VNCTray"
 RegConfig:   Registry changed
 Config:      List parameter SecurityTypes: Invalid value 'VeNCrypt'
 RegConfig:   Unable to process SecurityTypes
 Config:      Set QueryConnect(Bool) to on
 Config:      Set QueryOnlyIfLoggedOn(Bool) to off
 Config:      Set PortNumber(Int) to 5900
 Config:      Set IdleTimeout(Int) to 0
 Config:      Set LocalHost(Bool) to on
 Config:      Set Hosts(String) to +,
 Config:      Set AcceptKeyEvents(Bool) to on
 Config:      Set RawKeyboard(Bool) to off
 Config:      Set AcceptPointerEvents(Bool) to on
 Config:      Set AcceptCutText(Bool) to off
 Config:      Set SendCutText(Bool) to off
 Config:      Set DisableLocalInputs(Bool) to off
 Config:      Set DisconnectClients(Bool) to on
 Config:      Set AlwaysShared(Bool) to off
 Config:      Set NeverShared(Bool) to off
 Config:      Set DisconnectAction(Enum) to None
 Config:      Set RemoveWallpaper(Bool) to on
 Config:      Set DisableEffects(Bool) to on
 Config:      Set UpdateMethod(Int) to 0
 Config:      Set PollConsoleWindows(Bool) to on
 Config:      Set UseCaptureBlt(Bool) to on
 RegConfig:   Unable to process UseHooks
 Config:      Set Protocol3.3(Bool) to off
 Config:      Set Password(Binary)
 ManagedListener: Set server to 00000062e95fef00
 ManagedListener: Set port to 5900
 MsgWindow:   Null _this in 00000000001304dc, message 24
 MsgWindow:   Null _this in 00000000001304dc, message 81
 ManagedListener: Set filter to +,
 VNCServerWin32: Refreshing tray icon
 VNCServerWin32: Entering message loop
 MsgWindow:   Null _this in 00000000001304dc, message 83
 MsgWindow:   Created window "VNCTray" (00000000001304dc)
```
initiating the connection:
```
Sun Jun 08 02:39:32 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::58289
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSVnc (258)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000021946e65d20)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000021946e65d20)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)
```
entering the passphrase:
```
Sun Jun 08 02:39:42 2025
 SConnection: Processing security message
 VNCServerST: Starting desktop
 SDisplay:    Starting
 SDisplay:    Attaching to virtual desktop
 SDisplay:    Deleting old pixel buffer & device
 SDisplay:    Creating pixel buffer
 DeviceContext: 32-bit BitFields: (ff0000, ff00, ff)
 DIBSectionBuffer: recreateBuffer()
 MsgWindow:   Creating window "rfb::win32::SDisplayCorePolling"
 MsgWindow:   Null _this in 00000000001a045e, message 24
 MsgWindow:   Null _this in 00000000001a045e, message 81
 MsgWindow:   Null _this in 00000000001a045e, message 83
 MsgWindow:   Created window "rfb::win32::SDisplayCorePolling"
              (00000000001a045e)
 SDisplayCorePolling: setScreenRect
 SDisplay:    Started Polling
 MsgWindow:   Creating window "WMMonitor"
 MsgWindow:   Null _this in 000000000006072a, message 24
 MsgWindow:   Null _this in 000000000006072a, message 81
 MsgWindow:   Null _this in 000000000006072a, message 83
 MsgWindow:   Created window "WMMonitor" (000000000006072a)
 MsgWindow:   Creating window "Clipboard"
 MsgWindow:   Null _this in 00000000003606f0, message 24
 MsgWindow:   Null _this in 00000000003606f0, message 81
 MsgWindow:   Null _this in 00000000003606f0, message 83
 MsgWindow:   Created window "Clipboard" (00000000003606f0)
 Clipboard:   Local clipboard changed by 0000000000000000
 Clipboard:   No clipboard notifier registered
 Clipboard:   Registered clipboard handler
 SInput:      Found dead key 0x60 '`'
 SInput:      Found dead key 0xb4 '�'
 SInput:      Found dead key 0x5e '^'
 CleanDesktop: Disable desktop wallpaper/Active Desktop
 CleanDesktop: ActiveDesktop::enable
 CleanDesktop: Failed to get/set Active Desktop options: -2147467263
 CleanDesktop: Disable desktop effects
 SDisplay:    Started
 VNCSConnST:  Closing [::1]::58289: basic_string::_M_construct null not valid
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 Connections: Closed: [::1]::58289
 VNCServerST: Stopping desktop
 SDisplay:    Stopping
 ComparingUpdateTracker: 0 pixels in / 0 pixels out
 ComparingUpdateTracker: (1:nan ratio)
 SDisplay:    Stopping Polling
 MsgWindow:   Null _this in 00000000001a045e, message 2
 MsgWindow:   Null _this in 00000000001a045e, message 82
 MsgWindow:   Destroyed window "rfb::win32::SDisplayCorePolling"
              (00000000001a045e)
 MsgWindow:   Null _this in 000000000006072a, message 2
 MsgWindow:   Null _this in 000000000006072a, message 82
 MsgWindow:   Destroyed window "WMMonitor" (000000000006072a)
 Clipboard:   Removing 00000000003606f0 from chain (next is 00000000001004c2)
 MsgWindow:   Null _this in 00000000003606f0, message 2
 MsgWindow:   Null _this in 00000000003606f0, message 82
 MsgWindow:   Destroyed window "Clipboard" (00000000003606f0)
 CleanDesktop: Restore desktop effects
 Registry:    Duplicated ffffffff80000001 to ffffffff80000001
 Registry:    openKey(ffffffff80000001,Control Panel\Desktop,rw) =
              00000000000003ac
 Registry:    RegCloseKey(ffffffff80000001)
 Registry:    RegCloseKey(00000000000003ac)
 CleanDesktop: Restore desktop wallpaper
 SDisplay:    Stopped

Sun Jun 08 02:39:44 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::58290
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSVnc (258)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 000002194692bd80)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 000002194692bd80)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)
```
entering the passphrase:
```
Sun Jun 08 02:39:57 2025
 SConnection: Processing security message
 VNCServerST: Starting desktop
 SDisplay:    Starting
 SDisplay:    Attaching to virtual desktop
 SDisplay:    Deleting old pixel buffer & device
 SDisplay:    Creating pixel buffer
 DeviceContext: 32-bit BitFields: (ff0000, ff00, ff)
 DIBSectionBuffer: recreateBuffer()
 MsgWindow:   Creating window "rfb::win32::SDisplayCorePolling"
 MsgWindow:   Null _this in 00000000000f06ec, message 24
 MsgWindow:   Null _this in 00000000000f06ec, message 81
 MsgWindow:   Null _this in 00000000000f06ec, message 83
 MsgWindow:   Created window "rfb::win32::SDisplayCorePolling"
              (00000000000f06ec)
 SDisplayCorePolling: setScreenRect
 SDisplay:    Started Polling
 MsgWindow:   Creating window "WMMonitor"
 MsgWindow:   Null _this in 00000000001c045e, message 24
 MsgWindow:   Null _this in 00000000001c045e, message 81
 MsgWindow:   Null _this in 00000000001c045e, message 83
 MsgWindow:   Created window "WMMonitor" (00000000001c045e)
 MsgWindow:   Creating window "Clipboard"
 MsgWindow:   Null _this in 00000000003906f0, message 24
 MsgWindow:   Null _this in 00000000003906f0, message 81
 MsgWindow:   Null _this in 00000000003906f0, message 83
 MsgWindow:   Created window "Clipboard" (00000000003906f0)
 Clipboard:   Local clipboard changed by 0000000000000000
 Clipboard:   No clipboard notifier registered
 Clipboard:   Registered clipboard handler
 SInput:      Found dead key 0x60 '`'
 SInput:      Found dead key 0xb4 '�'
 SInput:      Found dead key 0x5e '^'
 CleanDesktop: Disable desktop wallpaper/Active Desktop
 CleanDesktop: ActiveDesktop::enable
 CleanDesktop: Failed to get/set Active Desktop options: -2147467263
 CleanDesktop: Disable desktop effects
 SDisplay:    Started
 VNCSConnST:  Closing [::1]::58290: basic_string::_M_construct null not valid
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 Connections: Closed: [::1]::58290
 VNCServerST: Stopping desktop
 SDisplay:    Stopping
 ComparingUpdateTracker: 0 pixels in / 0 pixels out
 ComparingUpdateTracker: (1:nan ratio)
 SDisplay:    Stopping Polling
 MsgWindow:   Null _this in 00000000000f06ec, message 2
 MsgWindow:   Null _this in 00000000000f06ec, message 82
 MsgWindow:   Destroyed window "rfb::win32::SDisplayCorePolling"
              (00000000000f06ec)
 MsgWindow:   Null _this in 00000000001c045e, message 2
 MsgWindow:   Null _this in 00000000001c045e, message 82
 MsgWindow:   Destroyed window "WMMonitor" (00000000001c045e)
 Clipboard:   Removing 00000000003906f0 from chain (next is 0000000000000000)
 MsgWindow:   Null _this in 00000000003906f0, message 2
 MsgWindow:   Null _this in 00000000003906f0, message 82
 MsgWindow:   Destroyed window "Clipboard" (00000000003906f0)
 CleanDesktop: Restore desktop effects
 Registry:    Duplicated ffffffff80000001 to ffffffff80000001
 Registry:    openKey(ffffffff80000001,Control Panel\Desktop,rw) =
              00000000000003a4
 Registry:    RegCloseKey(ffffffff80000001)
 Registry:    RegCloseKey(00000000000003a4)
 CleanDesktop: Restore desktop wallpaper
 SDisplay:    Stopped

Sun Jun 08 02:39:59 2025
 SocketManager: Accepting incoming connection
 Connections: Accepted: [::1]::58291
 SConnection: Reading protocol version
 SConnection: Client needs protocol version 3.8
 SConnection: Processing security type message

Sun Jun 08 02:40:00 2025
 SConnection: Processing security type message
 SConnection: Client requests security type VeNCrypt(19)
 SConnection: Processing security message
 SConnection: Processing security message
 SConnection: Processing security message
 SVeNCrypt:   Client requests security type TLSVnc (258)
 TLS:         Process security message (session 0000000000000000)
 TLS:         Anonymous session has been set
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000021946912530)
 TLS:         Deferring completion of TLS handshake: Resource temporarily
              unavailable, try again.
 SConnection: Processing security message
 TLS:         Process security message (session 0000021946912530)
 TLS:         TLS handshake completed with
              (TLS1.2)-(ANON-ECDH)-(AES-256-CBC)-(SHA1)
```
exiting the server:
```
Sun Jun 08 02:40:21 2025
 VNCServerWin32: Server exited cleanly
 Registry:    RegCloseKey(0000000000000254)
 VNCServerST: Shutting down server FERMI
 VNCSConnST:  Closing [::1]::58291: Server shutdown
 EncodeManager: Framebuffer updates: 0
 EncodeManager:   Total: 0 rects, 0 pixels
 EncodeManager:          0 B (1:nan ratio)
 main:        WinVNC service destroyed
 main:        WinVNC process quitting
```

**Comment by calestyo** [comment]
Could it be that 139f0e8a4b81fe3dcd9476ec5ee16ea5e74af901 is the offender? Something that doesn't play nice with `nullptr`?

Though that could at beast explain why it doesn't work anymore at all with > 1.13.1 ... not why it worked only at the 2nd try there.

**Comment by CendioOssman** [comment]
> Could it be that [139f0e8](https://github.com/TigerVNC/tigervnc/commit/139f0e8a4b81fe3dcd9476ec5ee16ea5e74af901) is the offender? Something that doesn't play nice with `nullptr`?

No, that's just a syntax change. I'm guessing the cause is one of b99daadb05e14e85da6c5e905057e10fc27c0fcf or adaedc9629c5f4b124a1872bac6a11844b7f576e. `std::string` doesn't like null pointers. We fixed a similar issue in 583c11fc55a64ed1ebdd8e6bc11c05cce004a5b8.

**Comment by CendioOssman** [comment]
This might be the issue:

```diff
diff --git a/win/winvnc/QueryConnectDialog.cxx b/win/winvnc/QueryConnectDialog.cxx
index 5d6098980..9a95e23ed 100644
--- a/win/winvnc/QueryConnectDialog.cxx
+++ b/win/winvnc/QueryConnectDialog.cxx
@@ -48,7 +48,8 @@ QueryConnectDialog::QueryConnectDialog(network::Socket* sock_,
                                        const char* userName_,
                                        VNCServerWin32* s)
 : Dialog(GetModuleHandle(nullptr)),
-  sock(sock_), peerIp(sock->getPeerAddress()), userName(userName_),
+  sock(sock_), peerIp(sock->getPeerAddress()),
+  userName(userName_?userName_:""),
   approve(false), server(s) {
 }
 
```

Are you able to do your own build to test this?

**Comment by CendioOssman** [comment]
Building using MSYS2 is pretty easy. You should be able to just copy what we do for GitHub actions:

https://github.com/TigerVNC/tigervnc/blob/b93fe2924acfb2b719033aa3511b4831412a05fc/.github/workflows/build.yml#L44-L70

I do that regularly for vncviewer.

**Comment by calestyo** [comment]
> Building using MSYS2 is pretty easy. 

Not so quite ;-)

I guess Github must be doing some more magic there, which was rather a pain to fiddle out manually.
- merely installing the `pacman` packages from above gives some dependency errors for `mingw-w64-x86_64-gnutls` and `mingw-w64-x86_64-pixman` (some `mingw-*-gcc-libs` packages or so)... I forced installation with `-dd`... either that just worked or a later `packman -S mingw-w64-x86_64-toolchin` helped
- then regular `cmake` doesn`t work, one needs the mingw cmake package
- then it took me to realise that one needs to add `/mingw64/bin` to the `PATH` or no compiler, etc is found.
- even with that, `make` still fails, but only after it has build `winvnc4.exe` and during the tests... it did run through when I used `-DBUILD_SHARED_LIBS=OFF` on `cmake`... not sure if that was really the cure
- the created `winvnc4.exe` cannot be started from windows itself... complins about all kinds of missing DLLs (libgcc_s_seh-1.dll, libgmp*,  libhogweed*, libnettle*
- **but** it can be run from within the MSYS2 shell, after which it opens a console window with log output... no icon in the systray

But now to the interesting stuff. I've tested on ae155957943d68f1b5346d5193d381e772aac3ce.

With your patch from above, the problem seems fixed (at least at that manually run from MSYS2 exe).

Compiling it again, without your patch, the prompt window doesn't show up again (also not after retries, at which it worked with 1.13.1).<br>
However, that "short dimming" of the screen, when it disables the wallpaper and effects, and which I've still seen in the official 1.15.0, seem gone in my own compiled version (without your patch - haven't looked when I had the version with).


Nevertheless... looks like your patch would fix it... so please commit and release ;-)

Thanks,
Chris.

**Comment by calestyo** [comment]
> That's because we are now more strict about valid values to parameters. So your issue is seen here in the log:
> Config:      List parameter SecurityTypes: Invalid value 'VeNCrypt'
>
> I'm guessing this is a remnant from some development version as I don't think we ever shipped a version where this value was needed.

Hmm I've removed it manually from both HKEY_LOCAL_MACHINE and HKEY_CURRENT_USER, after that, when I select just None/None in the GUI, it stays at that (when restarting the conf tool), but it still brings the `VeNCrypt` back in the registry... no idea what's going on ^^

Also, it still seems as I I could no longer log in without a password, even if the GUI shows None enabled.

**Comment by CendioOssman** [comment]
> Nevertheless... looks like your patch would fix it... so please commit and release ;-)

Thanks for testing. Added as 419be7dcb55e29a3be22a7cb2793b8434107361a.

> Hmm I've removed it manually from both HKEY_LOCAL_MACHINE and HKEY_CURRENT_USER, after that, when I select just None/None in the GUI, it stays at that (when restarting the conf tool), but it still brings the `VeNCrypt` back in the registry... no idea what's going on ^^
> 

Odd. It should never be adding that to the list of configured types. But WinVNC isn't as heavily tested, so there might be some subtle bug hiding somewhere. :/

> Also, it still seems as I I could no longer log in without a password, even if the GUI shows None enabled.

That is likely related. The default is `TLSVnc, VncAuth` after all.

**Comment by calestyo** [comment]
The patch fixes it indeed. With it, `VeNCrypt` is no longer unconditionally added. And without that, password-less login works again, too (it does so already when one manually removes the value from the registry).

So go ahead and merge :-) Thanks!

**Comment by calestyo** [comment]
With issue I mean, that the first connection doesn't show the "prompt dialog" on the server side and the connection times out, but it works then on subsequent connections.

**Comment by CendioOssman** [comment]
We'd need a new debug log from at least the server. But please include the client as well if you can.

**Comment by calestyo** [comment]
Took me a while to get access to the test system again (right now I'd have so until Sunday).

I've first made a full uninstall and then installed the 64bit version of 1.16.0.

Now the thing is:
When I use server mode it actually works (the first time already).

But when I user service mode it still fails (only when connecting the first time after the service was restarted).

I tried to get a debug log from the service, basically by adding `-Log *:file:100` to the `ImagePath` key in `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\TigerVNC`.

With that it actually creates the log file (deleted it first to be sure), but it doesn't actually debug log anything, it merely prints:
```

Thu Feb 19 22:15:17 2026
 Service:     Service
```

Any idea how to get it to debug log something in service mode? Or does it perhaps already ring a bell that the patch you made earlier, needs something in addition for service mode?


Thanks,
Chris

**Comment by CendioOssman** [comment]
Logging is a bit different in service mode. There is this note in the code:

```c++
  // - Set the service-mode logging defaults
  //   These will be overridden by the Log option in the
  //   registry, if present.
  logParams.setParam("*:EventLog:0,Connections:EventLog:100");
```

So it looks like you need to modify the `Log` entry in `HKEY_LOCAL_MACHINE\Software\TigerVNC\WinVNC4`.

Should probably check for `VeNCrypt` among the settings there as well.

**Comment by calestyo** [comment]
Here we go

First, with "prompt user for confirmation" enabled in TigerVNC server:
[WinVNC4.with-tigervnc-user-prompt.log](https://github.com/user-attachments/files/25463167/WinVNC4.with-tigervnc-user-prompt.log)
[remmina.with-tigervnc-user-prompt.log](https://github.com/user-attachments/files/25463168/remmina.with-tigervnc-user-prompt.log)

And just in case it should be needed, with "prompt user for confirmation" disabled in TigerVNC server (there it always works on the first try):
[WinVNC4.without-tigervnc-user-prompt.log](https://github.com/user-attachments/files/25463170/WinVNC4.without-tigervnc-user-prompt.log)
[remmina.without-tigervnc-user-prompt.log](https://github.com/user-attachments/files/25463171/remmina.without-tigervnc-user-prompt.log)


Client in both cases remmina 1.4.40+dfsg-2 from Debian. Server in both cases 1.16.0 on Win11 64bit.

**Comment by CendioOssman** [comment]
Nothing obvious in the log, unfortunately. It looks like the dialog is simply ignored and times out after 10 seconds.

Have you verified if it still works with the version of TigerVNC you upgraded from? It might be some other change that is messing with our dialog being shown. The fact that it only breaks in service mode does suggest some permission issue.

**Comment by calestyo** [comment]
> Nothing obvious in the log, unfortunately. It looks like the dialog is simply ignored and times out after 10 seconds.

Yep... one even sees that Windows removes the wallpaper.... just as already was for the server mode (but there it's fixed since your patch from [above](https://github.com/TigerVNC/tigervnc/issues/1644#issuecomment-2955107121)).

> Have you verified if it still works with the version of TigerVNC you upgraded from? It might be some other change that is messing with our dialog being shown. The fact that it only breaks in service mode does suggest some permission issue.

I wouldn't even recall a version where it worked (under win11)... looking a my original report here, I wrote that it already failed with 1.12.x.

**Comment by calestyo** [comment]
btw:
You might want to add:
> So it looks like you need to modify the Log entry in HKEY_LOCAL_MACHINE\Software\TigerVNC\WinVNC4.

to https://github.com/TigerVNC/tigervnc/wiki/Debug-Logs#windows

**Comment by CendioOssman** [comment]
> 
> I wouldn't even recall a version where it worked (under win11)... looking a my original report here, I wrote that it already failed with 1.12.x.

Oh? I thought this was a regression from somewhere after the last fix.

In that case, I'd like you to file a new issue instead to keep things clear. It might be related to #516.

> You might want to add:
> 
> > So it looks like you need to modify the Log entry in HKEY_LOCAL_MACHINE\Software\TigerVNC\WinVNC4.
> 
> to https://github.com/TigerVNC/tigervnc/wiki/Debug-Logs#windows

Good idea. I'll try to get that updated.

**Comment by calestyo** [comment]
Well it *did* use to work so it is a regression... I just don't recall in which version it still worked ;-)

And it's literally the same problem that this issue was originally about (except that it's now fixed for server mode).
But anyway, if you prefer I can still open a new ticket.
